azure-quantum/tests/unit/test_target.py

```diff
@@ -79,7 +80,72 @@ def _test_job_submit_ionq(self, num_shots):
             assert "histogram" in results
             assert results["histogram"]["0"] == 0.5
             assert results["histogram"]["7"] == 0.5
+
             if num_shots:
                 assert job.details.input_params.get("shots") == num_shots
             else:
                 assert job.details.input_params.get("shots") is None
+
+
+class TestHoneywell(QuantumTestBase):
+    mock_create_job_id_name = "create_job_id"
+    create_job_id = Job.create_job_id
+
+    def get_test_job_id(self):
+        return ZERO_UID if self.is_playback \
+               else Job.create_job_id()
+
+    def _teleport(self):
+        return """OPENQASM 2.0;
+        include "qelib1.inc";
+
+        qreg q[3];
+        creg c0[1];
+        creg c1[3];
+
+        h q[0];
+        cx q[0], q[1];
+        x q[2];
+        h q[2];
+        cx q[2], q[0];
+        h q[2];
+        measure q[0] -> c1[0];
+        c0[0] = c1[0];
+        if (c0==1) x q[1];
+        c0[0] = 0;
+        measure q[2] -> c1[1];
+        c0[0] = c1[1];
+        if (c0==1) z q[1];
+        c0[0] = 0;
+        h q[1];
+        measure q[1] -> c1[2];
+        """
+
+    def test_job_submit_honeywell(self):
+
+        with unittest.mock.patch.object(
+            Job,
+            self.mock_create_job_id_name,
+            return_value=self.get_test_job_id(),
+        ):
+            workspace = self.create_workspace()
+            circuit = self._teleport()
+            target = Honeywell(workspace=workspace)
+            job = target.submit(circuit)
+
+            # Make sure the job is completed before fetching the results
+            # playback currently does not work for repeated calls
+            if not self.is_playback:
+                self.assertEqual(False, job.has_completed())
+                if self.in_recording:
+                    import time
+                    time.sleep(3)
+                job.refresh()
+                job.wait_until_completed()
```

Comment:
I don't think we should wait for completion; unfortunately honeywell system is not mature enough that they can guarantee that their targets will have 99.99% availability, not even their api validator.

Reply:
This is meant for when we are recording playback for the API. In that case, we do want to wait until it is completed.
If we are not recording and running this in CI, we can set a timeout and check the status of the job later.

Reply:
I'm merging this to the feature branch for now, let's loop back on the e2e scenario and how to test it when we are ready.